### PR TITLE
Up-NRHM reports - Added possibility to change language in report

### PR DESCRIPTION
**Pull request by lwyszomi**
@snopoke @czue 
http://manage.dimagi.com/default.asp?171208#985324

### CODE REVIEW

File: custom/up_nrhm/filters.py [hunk 1]
```diff
@@ -128,3 +128,12 @@ class NRHMDatespanFilter(DatespanFilter):
 
 class NRHMDatespanMixin(DatespanMixin):
     datespan_field = NRHMDatespanFilter
+
+class LanguageFilter(BaseSingleOptionFilter):
```
Comment (dimagimon) [inline custom/up_nrhm/filters.py]: E302 expected 2 blank lines, found 1

File: custom/up_nrhm/reports/asha_facilitators_report.py [hunk 2]
```diff
@@ -3,11 +3,11 @@
 from corehq.apps.reports.standard import CustomProjectReport, DatespanMixin
 from corehq.apps.reports.util import format_datatables_data
 from custom.up_nrhm.sql_data import ASHAFacilitatorsData
-from django.utils.translation import ugettext_lazy as _, ugettext_noop
+from django.utils.translation import ugettext as _
 
 
 class ASHAFacilitatorsReport(GenericTabularReport, DatespanMixin, CustomProjectReport):
-    name = ugettext_noop("Format-2 Consolidation of the Functionality numbers")
+    name = _("Format-2 Consolidation of the Functionality numbers")
```
Comment (czue) [inline custom/up_nrhm/reports/asha_facilitators_report.py]: this should be `ugettext_lazy` - http://commcare-hq.readthedocs.org/en/latest/translations.html#tagging-strings-in-views

File: custom/up_nrhm/reports/asha_functionality_checklist_report.py [hunk 3]
```diff
@@ -4,11 +4,11 @@
 from custom.up_nrhm.filters import NRHMDatespanMixin
 from custom.up_nrhm.sql_data import ASHAFunctionalityChecklistData, ASHAAFChecklistData
 from dimagi.utils.dates import force_to_datetime
-from django.utils.translation import ugettext_lazy as _, ugettext_noop
+from django.utils.translation import ugettext as _
 
 
 class ASHAFunctionalityChecklistReport(GenericTabularReport, NRHMDatespanMixin, CustomProjectReport):
-    name = ugettext_noop("Format-1 for ASHA Sanginis")
+    name = _("Format-1 for ASHA Sanginis")
```
Comment (czue) [inline custom/up_nrhm/reports/asha_functionality_checklist_report.py]: same

File: custom/up_nrhm/reports/asha_reports.py [hunk 4]
```diff
@@ -21,8 +24,10 @@ def total_rows(report):
     return {}
 
 
-class ASHAReports(GenericTabularReport, NRHMDatespanMixin, CustomProjectReport):
-    fields = [SampleFormatFilter, NRHMDatespanFilter, DrillDownOptionFilter, ASHAMonthFilter, YearFilter]
+class ASHAReports(GenericTabularReport, NRHMDatespanMixin, CustomProjectReport, LangMixin):
+    fields = [SampleFormatFilter, LanguageFilter,
+              NRHMDatespanFilter, DrillDownOptionFilter,
+              ASHAMonthFilter, YearFilter]
     name = _("ASHA Sangini Reports")
```
Comment (czue) [inline custom/up_nrhm/reports/asha_reports.py]: same

**Comment by czue** [comment]
not going to go through and mark everything up, but read through the documentation I linked, particularly: "`ugettext` should be used in code that will be run per-request. `ugettext_lazy` should be used in code that is run at module import." and then let me know when you've had a chance to update this

File: custom/up_nrhm/reports/asha_facilitators_report.py (hunk 2 above)
Comment (snopoke) [inline custom/up_nrhm/reports/asha_facilitators_report.py]: Had the same discussion here: https://github.com/dimagi/commcare-hq/pull/7669/files#r35741689

Apparently it doesn't work

Comment (czue) [inline custom/up_nrhm/reports/asha_facilitators_report.py]: strange - though in this case I think the right thing would be `ugettext_noop` not `ugettext`.

**Comment by kkrampa** [comment]
@czue Lukasz is on vacation. Change this to ugettext causes problem that Simon mentioned.

**Comment by czue** [comment]
@snopoke fixed the class-level references

**Pull request merged**